### ISSUE TITLE
Define "text content"

### ISSUE THREAD

**Issue by annethyme** [act-rules/act-rules.github.io#136]

**Comment by annethyme** [comment]
Shoul take into account this conversation: https://github.com/auto-wcag/auto-wcag/issues/104

### Wilco: 
@kasperisager My point is that the "X" in <button aria-label="close">X</button> isn't text as WCAG defines it. It's actually non-text content. If you replaced that "X" with an image that looks identical, "X" would not be an appropriate alternative text.

This button example is a user interface component without text. Since the SC is only applicable to "user interface components with labels that include text" - this case should not fail the rule.

### Kasper: 
That's a very valid point, thanks for the clarification! A lot of rules will however depend on the presence of text in the WCAG sense, so we will need to define it in a way that can be implemented in code if we wan't to automate such rules, this one included.

Edit: We should also check how AT deals with text that isn't human language. Does AT care or does it treat symbolic characters, such as an X, differently?

**Comment by kasperisager** [comment]
There's also the case of punctuation. We just encountered something similar to this at Siteimprove for our contrast check:

```html
<div class="bad-contrast">
  <span class="good-contrast">This is text</span>
  .
<div>
```

We were flagging `.bad-contrast` as it contained a period with insufficient contrast, which might not be entirely fair.

**Comment by annethyme** [comment]
In the old "text content" definition, the accessible name for non-text content is also considered part of the text content: https://github.com/auto-wcag/auto-wcag/blob/master/pages/algorithms/text-content.md

@WilcoFiers, do you feel that this should still be a part of the text content definition? 

I guess if we go in this direction, we need to look at the accessible name also of the element itself - which might even have overwritten the actual text content of the element...

**Comment by jkodu** [comment]
Please validate this issue.

**Comment by jkodu** [comment]
Have a glossary for the same - https://act-rules.github.io/glossary/#text-content
